contracts/helper/RebalancingSetEthBidder.sol

```diff
@@ -0,0 +1,236 @@
+import { IRebalanceAuctionModule } from "../core/interfaces/IRebalanceAuctionModule.sol";
+import { IRebalancingSetToken } from "../core/interfaces/IRebalancingSetToken.sol";
+import { ITransferProxy } from "../core/interfaces/ITransferProxy.sol";
+import { IWETH } from "../lib/IWETH.sol";
+
+import { ERC20Wrapper } from "../lib/ERC20Wrapper.sol";
+
+
+/**
+ * @title RebalancingSetEthBidder
+ * @author Set Protocol
+ *
+ * A helper contract that wraps your ETH into WETH for bidding in the RebalanceAuctionModule.
+ */
+contract RebalancingSetEthBidder is
+    ReentrancyGuard
+{
+    using SafeMath for uint256;
+    
+    // Address and instance of RebalanceAuctionModule contract
+    IRebalanceAuctionModule public rebalanceAuctionModule;
+
+    // Address and instance of RebalanceAuctionModule contract
+    ITransferProxy public transferProxy;
+
+    // Address and instance of Wrapped Ether contract
+    IWETH public weth;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * Constructor function for RebalancingSetEthBidder
+     *
+     * @param _rebalanceAuctionModule   The address of RebalanceAuctionModule
+     * @param _transferProxy            The address of TransferProxy
+     * @param _weth                     The address of Wrapped Ether
+     */
+    constructor(
+        IRebalanceAuctionModule _rebalanceAuctionModule,
+        ITransferProxy _transferProxy,
+        IWETH _weth
+    )
+        public
+    {
+        // Commit the RebalanceAuctionModule instance
+        rebalanceAuctionModule = _rebalanceAuctionModule;
+
+        // Commit the TransferProxy instance
+        transferProxy = _transferProxy;
+
+        // Commit the WETH instance
+        weth = _weth;
+
+        // Add approvals of Wrapped Ether to the Transfer Proxy
+        ERC20Wrapper.approve(
+            address(_weth),
+            address(_transferProxy),
+            CommonMath.maxUInt256()
+        );
+    }
+
+    /**
+     * Fallback function. Disallows ether to be sent to this contract without data except when
+     * unwrapping WETH.
+     */
+    function ()
+        external
+        payable
+    {
+        require(
+            msg.sender == address(weth),
+            "RebalancingSetEthBidder.fallback: Cannot receive ETH directly unless unwrapping WETH"
+        );
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * Bid on rebalancing a given quantity of sets held by a rebalancing token wrapping or unwrapping
+     * any ETH involved. The tokens are returned to the user.
+     *
+     * @param  _rebalancingSetToken    Address of the rebalancing token being bid on
+     * @param  _quantity               Number of currentSets to rebalance
+     * @param  _allowPartialFill       Set to true if want to partially fill bid when quantity
+                                       is greater than currentRemainingSets
+     */
+
+    function bidAndWithdrawWithEther(
+        address _rebalancingSetToken,
```

Comment:
Knit: type `IRebalancingSetToken`

Reply:
Might keep it `address` since rebalanceAuctionModule.bidAndWithdraw takes in address as type

```
        // Bid in auction
        rebalanceAuctionModule.bidAndWithdraw(
            _rebalancingSetToken,
            _quantity,
            _allowPartialFill
        );```